### PR TITLE
Add visual debugging for entities

### PR DESCRIPTION
**Pull request by kokosha**
- Display rectangles and bounding boxes for the sprites of each entity.
- Highlight the center of each bounding box and create an axis to facilitate alignment verification.
- Implement a mode where billboards always faces the current camera.
- Reduce the billboard size for the player entity.
- Refactor the code to reuse existing functions.
- Convert sRGB to linear color inside AABB/Circle/Rectangle pipeline.

### CODE REVIEW

**Comment by hasenbanck** [comment]
She shader looks like it's a general purpose "rectangle" shader, like the AABB and circle shader. We could then use it the same way we already use the other two debug shader.

File: korangar/src/graphics/passes/forward/rectangle.rs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        if self.draw_count == 0 {
+            return;
+        }
+
+        self.instance_data.clear();
+
+        for instruction in instructions.rectangles.iter() {
+            self.instance_data.push(InstanceData {
+                world: instruction.world.into(),
+                color: instruction.color.into(),
```
Comment (hasenbanck) [inline korangar/src/graphics/passes/forward/rectangle.rs:189]:
```suggestion
                color: instruction.color.components_linear(),
```

This was already wrong in the Circle/AABB code. So it should be updated there too. The "Color" values are curently in sRGB and they need to be converted the pre-multiplied alpha, linear colors.

File: korangar/src/loaders/animation/mod.rs [hunk 2]
```diff
@@ -258,14 +266,50 @@ impl AnimationLoader {
 
                 return Vector2::new(size_x, size_y);
             }
+            #[cfg(feature = "debug")]
```
Comment (hasenbanck) [inline korangar/src/loaders/animation/mod.rs:269]: Please add a free line before this. Also: Do we need thos function to be inlined? Can't they be a free standing function?

File: korangar/src/main.rs [hunk 3]
```diff
@@ -2086,8 +2094,19 @@ impl Client {
...
+            let entity_camera = match true {
+                #[cfg(feature = "debug")]
+                _ if self.render_settings.get().show_entities_paper => &self.player_camera,
+                _ => current_camera,
+            };
+
             #[cfg_attr(feature = "debug", korangar_debug::debug_condition(render_settings.show_entities))]
-            self.map.render_entities(&mut self.entity_instructions, entities, current_camera);
+            self.map.render_entities(&mut self.entity_instructions, entities, entity_camera);
+            #[cfg(feature = "debug")]
```
Comment (hasenbanck) [inline korangar/src/main.rs:2105]: Please add a free line before this.

File: korangar/src/world/animation/mod.rs [hunk 4]
```diff
@@ -154,4 +182,41 @@ impl AnimationData {
...
+        &self,
+        instructions: &mut Vec<DebugRectangleInstruction>,
+        camera: &dyn Camera,
+        entity_position: Point3<f32>,
+        animation_state: &AnimationState,
+        head_direction: usize,
+        color_external: Color,
+        color_internal: Color,
+    ) {
+        const TILE_SIZE: f32 = 10.0;
```
Comment (hasenbanck) [inline korangar/src/world/animation/mod.rs:197]: Do we need to repreat the constants here? Can't we move them to the top of the file and re-use them?

**Review by hasenbanck — CHANGES_REQUESTED** [review]

File: korangar/src/graphics/passes/forward/rectangle.rs (hunk 1 above)
Comment (kokosha) [reply korangar/src/graphics/passes/forward/rectangle.rs:189]: Does I make the change in Circle/AABB code also?

File: korangar/src/loaders/animation/mod.rs (hunk 2 above)
Comment (kokosha) [reply korangar/src/loaders/animation/mod.rs:269]: They can be free standing, just not sure where to put the function.

File: korangar/src/world/animation/mod.rs (hunk 4 above)
Comment (kokosha) [reply korangar/src/world/animation/mod.rs:197]: I will put on top of the file

File: korangar/src/graphics/passes/forward/rectangle.rs (hunk 1 above)
Comment (hasenbanck) [reply korangar/src/graphics/passes/forward/rectangle.rs]: That would be great!

**Comment by hasenbanck** [comment]
Please squash, rebase and use the updated title of the issue for the commit message once you are finished!

**Comment by kokosha** [comment]
I change the variable naming also for better readability and will rebase on top of it.

**Comment by hasenbanck** [comment]
LGTM

**Review by hasenbanck — APPROVED** [review]

**Pull request merged**